db/db_iter.cc

```diff
@@ -82,6 +82,7 @@ class DBIter: public Iterator {
     if (kForward==direction_)
     {
       ParsedInternalKey parsed;
+      parsed.type=kTypeValue; parsed.sequence=0; parsed.expiry=0; // clear warnings
```

Comment:
... but the comment on the constructor says that they are not initializing there for performance reasons.  Upgrading comment in db_iter.cc.